std/algorithm/searching.d

```diff
@@ -2753,6 +2754,33 @@ if (isForwardRange!R1 && isForwardRange!R2
     assert(findSkip(s, "def") && s.empty);
 }
 
+/*
+* Params:
+*  haystack = The
+*   $(REF_ALTTEXT forward range, isForwardRange, std,range,primitives) to search
+*   in.
+*  pred = Custom predicate for comparison of haystack and needle
+*
+* Returns: $(D true) if pred was evaluated to true at the beginning,
+* in which case $(D haystack) is positioned after the end of
+* the first occurrence of $(D needle); otherwise $(D false),
+* leaving $(D haystack) untouched.
+*/
+bool findSkip(alias pred, R1)(ref R1 haystack)
+if (isForwardRange!R1 && is(typeof(unaryFun!pred(haystack.front))))
+{
+    return skipOver!(pred, R1)(haystack);
```

Comment:
Of course, if we decide to change the behavior introduced in #5543, we could easily inline #5543 here ...